From 8ac154aad2da2216e2e741f7cb8990d36e563412 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Tue, 26 Apr 2022 11:20:08 +0200
Subject: [PATCH 059/175] fix(LinstorSR): always set vdi_path in
 generate_config

If the volume of a generated config is not related to HTTP/NBD
and if we already have a path to the resource, the VDI path is never
written to the config. So the config can't be used to attach the VDI...

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LinstorSR.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/LinstorSR.py b/drivers/LinstorSR.py
index e2d3d78..1855e3d 100755
--- a/drivers/LinstorSR.py
+++ b/drivers/LinstorSR.py
@@ -1970,7 +1970,7 @@ class LinstorVDI(VDI.VDI):
                 if not available:
                     raise xs_errors.XenError('VDIUnavailable')
 
-                resp['vdi_path'] = self.path
+            resp['vdi_path'] = self.path
         else:
             # Axiom: DRBD device is present on at least one host.
             resp['vdi_path'] = '/dev/http-nbd/' + volume_name
-- 
2.45.2

